src/main/java/org/embulk/util/config/Version.java

```diff
@@ -0,0 +1,138 @@
+        } catch (final Throwable any) {  // It must not fail.
+            return com.fasterxml.jackson.core.Version.unknownVersion();
+        }
+    }
+
+    private static com.fasterxml.jackson.core.Version parseVersion(final String implementationVersion) {
+        final String[] versionAndSnapshotInfo = implementationVersion.split("-", 2);
+        final String version;
+        final String snapshotInfo;
+        if (versionAndSnapshotInfo.length == 1) {
+            version = versionAndSnapshotInfo[0];
+            snapshotInfo = null;
+        } else if (versionAndSnapshotInfo.length == 2) {
+            version = versionAndSnapshotInfo[0];
+            snapshotInfo = versionAndSnapshotInfo[1];
+        } else {
+            throw new IllegalArgumentException("Empty version string.");
+        }
+
+        final String[] versionSplit = version.split("\\.");
+        if (versionSplit.length == 1) {
+            return new com.fasterxml.jackson.core.Version(
+                    Integer.parseInt(versionSplit[0]),
+                    0,
+                    0,
+                    snapshotInfo,
+                    GROUP_ID,
+                    ARTIFACT_ID);
+        } else if (versionSplit.length == 2) {
+            return new com.fasterxml.jackson.core.Version(
+                    Integer.parseInt(versionSplit[0]),
+                    Integer.parseInt(versionSplit[1]),
+                    0,
+                    snapshotInfo,
+                    GROUP_ID,
+                    ARTIFACT_ID);
+        } else if (versionSplit.length == 3) {
+            return new com.fasterxml.jackson.core.Version(
+                    Integer.parseInt(versionSplit[0]),
+                    Integer.parseInt(versionSplit[1]),
+                    Integer.parseInt(versionSplit[2]),
+                    snapshotInfo,
+                    GROUP_ID,
+                    ARTIFACT_ID);
+        } else if (versionSplit.length == 0) {
+            throw new IllegalArgumentException("Empty version.");
+        } else {
+            throw new IllegalArgumentException("3+ digits in version.");
+        }
+    }
+
+    private static String getImplementationVersion() {
+        try {
+            return getManifestImplementationVersion();
+        } catch (final Throwable any) {
+            // Pass-through.
+        }
+
+        try {
+            // Trying to get a version string from system properties. It is almost just for testing.
+            return System.getProperty("org.embulk.embulk_util_config.version");
+        } catch (final Throwable any) {
+            // Pass-through.
+        }
+
+        return null;
+    }
+
+    private static String getManifestImplementationVersion() throws Exception {
+        final URL selfJarUrl = Version.class.getProtectionDomain().getCodeSource().getLocation();
+        if (selfJarUrl == null || !selfJarUrl.getProtocol().equals("file")) {
+            throw new Exception();
```

Comment:
Some message would be useful.

Reply:
For the same reason above. Even if it throws any Exception, that would not be used anywhere, and just `unknownVersion` is set.